maro/simulator/scenarios/data_center/business_engine.py

```diff
@@ -0,0 +1,320 @@
+
+    def _load_configs(self):
+        """Load configurations."""
+
+        # Update self._config_path with current file path.
+        self.update_config_root_path(__file__)
+        with open(os.path.join(self._config_path, "config.yml")) as fp:
+            self._conf = safe_load(fp)
+
+        self._delay_duration: int = self._conf["delay_duration"]
+        self._pm_amount: int = self._conf["pm_amount"]
+        self._pm_cap_cpu: int = self._conf["pm_cap_cpu"]
+        self._pm_cap_mem: int = self._conf["pm_cap_mem"]
+
+    def _init_machines(self):
+        """Initialize the physical machines based on the config setting. The PM id starts from 0."""
+
+        self._machines: List[PhysicalMachine] = [
+            PhysicalMachine(
+                id=i,
+                cap_cpu=self._pm_cap_cpu,
+                cap_mem=self._pm_cap_mem
+            ) for i in range(self._pm_amount)
+        ]
+
+    def step(self, tick: int):
+        """Push business to next step.
+
+        Args:
+            tick (int): Current tick to process.
+        """
+        self._tick = tick
+
+        # Update all live VMs CPU utilization.
+        self._update_vm_util()
+        # Update all PM CPU utilization.
+        self._update_pm_util()
+        # TODO
+        # Generate VM requirement events from data file.
+        # It might be implemented by a for loop to process VMs in each tick.
+        vm_requirement_event = self._event_buffer.gen_cascade_event(
+            tick=tick,
+            event_type=Events.REQUIREMENTS,
+            payload=None
+        )
+        self._event_buffer.insert_event(event=vm_requirement_event)
+
+    def get_metrics(self) -> DocableDict:
+        """Get current environment metrics information.
+
+        Returns:
+            DocableDict: Metrics information.
+        """
+
+        return DocableDict(
+            metrics_desc,
+            energy_consumption=self._energy_consumption,
+            success_requirements=self._successful_requirements,
+            failed_requirements=self._failed_requirements,
+            total_latency=self._total_latency
+        )
+
+    def _register_events(self):
+        # Register our own events and their callback handlers.
+        self._event_buffer.register_event_handler(event_type=Events.REQUIREMENTS, handler=self._on_vm_required)
+        self._event_buffer.register_event_handler(event_type=Events.FINISHED, handler=self._on_vm_finished)
+
+        # Generate decision event.
+        self._event_buffer.register_event_handler(event_type=MaroEvents.TAKE_ACTION, handler=self._on_action_received)
+
+    def _update_vm_util(self):
+        """Update all live VMs CPU utilization.
+
+        The length of VMs utilization series could be difference among all VMs,
+        because index 0 represents the VM's CPU utilization at the tick it starts.
+        """
+
+        for vm in self._live_vm.values():
+            vm.util_cpu = vm.get_util(cur_tick=self._tick)
```

Comment:
util a common word for another meaning, even it's utilization the uint is usually %, so can use vm.cpu_workload, vm.memory_workload?

Reply:
if the original data is the utilization, what is the base? the originally expected cpu?

Reply:
So all the util in BE is stored as %. 
For VM, the base is req_cpu. 
For PM, the base is cap_cpu.